### ISSUE TITLE
Trakt plugin does not work with new Plex Movie agent

### ISSUE THREAD

**Issue by rg9400** [trakt/Plex-Trakt-Scrobbler#567]
Seems like the plugin is no longer able to sync or scrobble anything related to movies after upgrading to the latest agent. Can anyone else confirm?

Is this something we can fix, since the developer does not seem to be active anymore? It will be a shame if not, the plugin was super useful.

**Comment by ourtv** [comment]
I can confirm. No longer syncs movies but will tv. I also have no clue how to fix but do love the implementation and use of this plugin.

**Comment by InternationalAdvice0** [comment]
Same here... Sad.

**Comment by jyggen** [comment]
It's because the new movie agent uses an internal Plex GUID to identify each movie instead of IMDb and TMDB IDs as before. This plugin, like most other plugins and 3rd-party tools, likely uses this ID to know which movie to scrobble on Trakt - but since this new GUID is unique for Plex, there's no exact way to match a movie in Plex with a movie on Trakt.

**Comment by rg9400** [comment]
To get IDs back into the new Plex Agent, please vote for this feature. Once this is done, the Trakt plugin will have to be modified since the IDs won't be in the guid anymore. The original dev has disappeared, but maybe someone can fork it to fix the problem. We have to wait for if and how they choose to add back the ID
https://forums.plex.tv/t/allow-access-to-external-provider-ids-for-media/619090

**Comment by jeff15110168** [comment]
ah hoping they fix this soon, they responded positively in the thread @rg9400 linked

![image](https://user-images.githubusercontent.com/9892738/89970721-64ba7d00-dc27-11ea-850e-5b60d034ea1b.png)

**Comment by rg9400** [comment]
It's added in the upcoming release. Who wants to take a stab at forking this to work with the new format?

https://forums.plex.tv/t/native-plex-agents-allow-access-to-external-provider-ids-for-media-eg-imdb-tmdb-tvdb/619090/130

**Comment by rg9400** [comment]
New update is out. IDs show in the XML if you refresh metadata, like below. Does anyone feel comfortable forking this to use these new IDs so that it can work with the new agent

```xml
<Guid id="imdb://tt0892769"/>
<Guid id="tmdb://10191"/>
```

**Comment by rg9400** [comment]
I've done some changes to make this work with the new agent. They're hacks to override the `guid` logic since the codebase is so large. I also added the option for tvdb IDs in the Plex agent, so that it can theoretically handle the upcoming native agent. It does seem to be working with the new agent, at least the few things I tried.

My biggest concern is what might break due to these changes, specifically older/different agents that do not use the new external IDs. I also do not want to maintain an official fork of this plugin, so we need to see how to get the changes up so others can use them.

**Comment by dcbinder** [comment]
> This fork for the above PR has changes to make the plugin compatible with the new agent, at least per my testing. Both syncing and scrobbling should work now and be backwards compatible. TV scrobbling will work whenever they release the new native TV agent, but syncing would need further work. I might take a stab at it if it is similar to the movie sync stuff, otherwise, I'll leave it for another day.
> 
> This plugin has an absurdly large code base with tons of outdated packages and logic. I won't actually support it, just made something that should hopefully work for me and was willing to share with everyone else.
> 
> https://github.com/rg9400/Plex-Trakt-Scrobbler

How do you this plugin? For me the gear used to run it is not available anymore. 

Thanks

**Comment by rg9400** [comment]
> How do you run this plugin? For me the gear used to run it is not available anymore.
Thanks

Download the Trakttv.bundle from my fork, close down Plex, move the existing bundle from your plugin folder out (as a backup), then move my version in, restart Plex. The settings icon should still be visible (you can try Kitana to access the full menus or configure it from here http://trakt-for-plex.github.io/configuration/#/login).

If those are not working, then your plugin is not starting for some reason, and you need to find the logs in `Logs/PMS Plugin Logs`. It would be something like com.plexapp.trakttv or something. Some people have reported they needed to reinstall  Visual C++ Redistributable Packages for Visual Studio 2013 to get it to work.

Based on the error, I may be able to help, but it has multiple points of failure, so I can't guarantee anything.

**Comment by briancwelch** [comment]
While I did install your branch and did not have any issues with it loading, it still is not feeding my entire movie collection back to Trakt.  Anyone have more insight on this?

**Comment by jbreitbart** [comment]
@rg9400 Should I still use your fork or are all your changes from your fork now available in this repository as well?

**Comment by rg9400** [comment]
all changes have been merged into this repo @jbreitbart

**Comment by jackmeat** [comment]
@rg9400 Works perfect, thank you so much. I just wanted to add, for the less skilled in this stuff like ME, I only needed to get a new pin from trakt and then it worked perfectly, no need for the reinstall as mentioned per your troubleshooting. Might help someone else in the future.

**Comment by SasSam** [comment]
> @rg9400 Works perfect, thank you so much. I just wanted to add, for the less skilled in this stuff like ME, I only needed to get a new pin from trakt and then it worked perfectly, no need for the reinstall as mentioned per your troubleshooting. Might help someone else in the future.

@jackmeat Does it work with TV shows as well? For me, it works only with movies. Why did you have to get a new pin from trakt btw?

**Comment by jackmeat** [comment]
It is working for TV shows as well, but I did run into a different issue a month or so ago where it began marking my watched shows as unwatched out of the blue. I have since turned off the push and pull, and have it set to push only to trakt which seems to be working fine now. I don't know why that only affected the tv shows and not movies, but I didn't report it. And I needed a new pin only to connect plex to trakt again via following the instructions @rg9400 had provided. I didn't need to contact trakt to do it, just a regular authorization pin. My wording made it sound like a chore. It wasn't

**Comment by jackmeat** [comment]
> > @rg9400 Works perfect, thank you so much. I just wanted to add, for the less skilled in this stuff like ME, I only needed to get a new pin from trakt and then it worked perfectly, no need for the reinstall as mentioned per your troubleshooting. Might help someone else in the future.
> 
> @jackmeat Does it work with TV shows as well? For me, it works only with movies. Why did you have to get a new pin from trakt btw?

It is working for TV shows as well, but I did run into a different issue a month or so ago where it began marking my watched shows as unwatched out of the blue. I have since turned off the push and pull, and have it set to push only to trakt which seems to be working fine now. I don't know why that only affected the tv shows and not movies, but I didn't report it. And I needed a new pin only to connect plex to trakt again via following the instructions @rg9400 had provided. I didn't need to contact trakt to do it, just a regular authorization pin. My wording made it sound like a chore. It wasn't
